File: frontend/types/investigation.ts
export interface Diagnosis {
  root_cause: string;
  explanation: string;
  fix: string;
  kubectl_commands: string[];
  prevention: string;
  confidence: number;
  ai_available: boolean;
}

export interface InvestigationSummary {
  total_issues: number;
  cluster_healthy: boolean;
  issues: string[];
  errors: string[];
}

export interface InvestigationRequest {
  namespace?: string;
  context?: string;
  investigation_id?: string;
}

export interface InvestigationResponse {
  status: string;
  investigation_id?: string;
  investigation?: Record<string, unknown>;
  summary?: InvestigationSummary;
  diagnosis?: Diagnosis;
}

File: backend/models/investigation.py
from typing import Optional

from pydantic import BaseModel

from models.diagnosis import Diagnosis


class InvestigationRequest(BaseModel):
    namespace: str = "all"
    context: Optional[str] = None
    investigation_id: Optional[str] = None


class InvestigationSummary(BaseModel):
    total_issues: int
    cluster_healthy: bool
    issues: list[str]
    errors: list[str] = []


class InvestigationResponse(BaseModel):
    status: str
    investigation_id: Optional[str] = None
    investigation: Optional[dict] = None
    summary: Optional[InvestigationSummary] = None
    diagnosis: Optional[Diagnosis] = None
